src/txn/txn_rollback_to_stable.c

```diff
@@ -1400,13 +1400,11 @@ __rollback_to_stable_check(WT_SESSION_IMPL *session)
      * okay to have a transaction in prepared state.
      */
     txn_active = __txn_user_active(session);
-#ifdef HAVE_DIAGNOSTIC
-    if (txn_active)
-        WT_TRET(__wt_verbose_dump_txn(session));
-#endif
 
-    if (txn_active)
+    if (txn_active){
+        WT_TRET(__wt_verbose_dump_txn(session));
```

Comment:
Is it ok to dump this all the time? `__rollback_to_stable_check` is called at the beginning of `__rollback_to_stable`.
I can see in the original ticket we always want to print it, I guess it is fine then.
We could also do the following:
```suggestion
#ifdef HAVE_DIAGNOSTIC
  if(txn_active) {
#else
  if(txn_active && WT_VERBOSE_ISSET(session, WT_VERB_RTS)) {
#endif
    WT_TRET(__wt_verbose_dump_txn(session));
  }
```
However, this means we need the verbose category `WT_VERB_RTS` to be set when `HAVE_DIAGNOSTIC` is not enabled.

Reply:
Would that still help with the original goal of being able to diagnose BFs such as BF-24403 more easily? Whether it's okay or not I guess would also depend on how many times you might expect rollback to stable to fail due to active transactions?

Reply:
It wouldn't be as much helpful as your changes. One nitpick though, I don't think we need `txn_active` anymore, we may just do `if(__txn_user_active(session))`.